cpp/src/arrow/util/sse-util.h

```diff
@@ -110,7 +110,12 @@ static inline uint32_t SSE4_crc32_u32(uint32_t crc, uint32_t v) {
 }
 
 static inline uint32_t SSE4_crc32_u64(uint32_t crc, uint64_t v) {
+#if defined(__MINGW32__) && !defined(__MINGW64__)
+  DCHECK(false) << "MinGW-w64 32-bit doesn't support _mm_crc32_u64()";
```

Comment:
Did you run the tests or are you only concerned about building? This is pretty much not the right solution.

Reply:
This is just fixing build error.
I'm working on test support on MSYS2 in separated issues.
I'll be back again after we can run tests on MSYS2.